eventapi.py

```diff
@@ -0,0 +1,56 @@
+import json
+from datetime import datetime
+from flask import request
+from flask_restful import Resource, abort
+
+from event import Event
+from eventvalidations import *
+
+def read_args(request):
+    return json.loads(request.data.decode('utf-8'))
+
+class EventAPI(Resource):
+    def get(self, e_id):
+        event = Event.get_by_id(e_id)
+        return vars(event)
+
+    def put(self, e_id):
+        event = Event.get_by_id(e_id)
+        data = read_args(request)
+        try:
+            if 'from_time' in data:
+                read_time(data['from_time'])
+            if 'to_time' in data:
+                read_time(data['to_time'])
+            if 'days' in data:
+                validate_days(data['days'])
+            event.update(data)
+            validate_times(event.from_time, event.to_time)
+            return vars(event), 201
+
+        except Exception as e:
```

Comment:
theres many more exceptions that can happen here, the validation could be wrong, or the update to the DB could have thrown an exception, the connection could have been lost, the update could have failed due to duplicate keys... etc..

Reply:
^^ still needs to be addressed